### PR TITLE
feat: Create configuration for macOS+Bazel build.

### PR DESCRIPTION
**Pull request by coryan**
Create driver scripts for macOS and Bazel. In this initial version
the integration tests are not executed. That will happen in a future
PR.

This commit also includes a draft script to build with CMake, but it
does not work yet. Or more accurately, we need to create at least
one .o file in the `spanner_client_testing` library before it works.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/297)

### CODE REVIEW

File: ci/kokoro/macos/build-bazel.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ $# != 1 ]]; then
```
Comment (devbww) [inline ci/kokoro/macos/build-bazel.sh]: Should probably use an arithmetic operator (i.e., `-ne`) here.

File: ci/kokoro/macos/build-cmake.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ $# != 3 ]]; then
```
Comment (devbww) [inline ci/kokoro/macos/build-cmake.sh]: Ditto.

File: ci/kokoro/macos/build-bazel.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    # unit tests do exactly that.
+    --action_env=GOOGLE_APPLICATION_CREDENTIALS="${GOOGLE_APPLICATION_CREDENTIALS}"
+    "--test_output=errors"
+    "--verbose_failures=true"
+    "--keep_going")
+if [[ -n "${BAZEL_CONFIG}" ]]; then
+    bazel_args+=("--config" "${BAZEL_CONFIG}")
+fi
+
+# The -DGRPC_BAZEL_BUILD is needed because gRPC does not compile on macOS unless
```
Comment (tmatsuo) [inline ci/kokoro/macos/build-bazel.sh]: Nit: the commnent seems outdated?

File: ci/kokoro/macos/build-cmake.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+readonly SOURCE_DIR="$2"
+readonly BINARY_DIR="$3"
+
+NCPU="$(sysctl -n hw.logicalcpu)"
+readonly NCPU
+
+source "${PROJECT_ROOT}/ci/colors.sh"
+
+echo
+echo "${COLOR_YELLOW}Starting docker build $(date) with $(nproc) cores${COLOR_RESET}"
```
Comment (tmatsuo) [inline ci/kokoro/macos/build-cmake.sh]: Nit: maybe remove?

File: ci/kokoro/macos/build-cmake.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  echo "Running the integration tests $(date)"
+  echo "================================================================"
+  source "${KOKORO_GFILE_DIR}/spanner-integration-tests-config.sh"
+  export GOOGLE_APPLICATION_CREDENTIALS="${KOKORO_GFILE_DIR}/spanner-credentials.json"
+  export GOOGLE_CLOUD_CPP_AUTO_RUN_EXAMPLES=yes
+
+  # Run the integration tests too.
+  env -C "${BINARY_DIR}" ctest \
+      -L integration-tests \
+      --output-on-failure -j "${NCPU}"
```
Comment (tmatsuo) [inline ci/kokoro/macos/build-cmake.sh]: Nit: Probably it will never be a problem, but we may want to use multiple jobs even on single processor machine.

Comment (coryan) [reply ci/kokoro/macos/build-cmake.sh]: Probably, these tests are likely to be I/O bound and running more than the number of cores is probably a good optimization, but we need to have more than 2 or 3 of them before this is a worthy optimization.

File: ci/kokoro/macos/build-bazel.sh (hunk 1 above)
Comment (coryan) [reply ci/kokoro/macos/build-bazel.sh]: Done.

File: ci/kokoro/macos/build-cmake.sh (hunk 2 above)
Comment (coryan) [reply ci/kokoro/macos/build-cmake.sh]: Done.

File: ci/kokoro/macos/build-bazel.sh (hunk 3 above)
Comment (coryan) [reply ci/kokoro/macos/build-bazel.sh]: Done.

File: ci/kokoro/macos/build-cmake.sh (hunk 4 above)
Comment (coryan) [reply ci/kokoro/macos/build-cmake.sh]: Done.

**Pull request merged**